res/controllers/Denon-MC7000-scripts.js

```diff
@@ -0,0 +1,859 @@
+    engine.makeConnection("[Channel4]", "VuMeter", MC7000.VuMeter);
+
+    // Platter Ring LED
+    midi.sendShortMsg(0x90, 0x64, MC7000.modeSingleLED);
+    midi.sendShortMsg(0x91, 0x64, MC7000.modeSingleLED);
+    midi.sendShortMsg(0x92, 0x64, MC7000.modeSingleLED);
+    midi.sendShortMsg(0x93, 0x64, MC7000.modeSingleLED);
+    engine.makeConnection("[Channel1]", "playposition", MC7000.JogLed);
+    engine.makeConnection("[Channel2]", "playposition", MC7000.JogLed);
+    engine.makeConnection("[Channel3]", "playposition", MC7000.JogLed);
+    engine.makeConnection("[Channel4]", "playposition", MC7000.JogLed);
+
+    // Vinyl mode LEDs
+    midi.sendShortMsg(0x90, 0x07, MC7000.isVinylMode ? 0x7F: 0x01);
+    midi.sendShortMsg(0x91, 0x07, MC7000.isVinylMode ? 0x7F: 0x01);
+    midi.sendShortMsg(0x92, 0x07, MC7000.isVinylMode ? 0x7F: 0x01);
+    midi.sendShortMsg(0x93, 0x07, MC7000.isVinylMode ? 0x7F: 0x01);
+
+    // PAD Mode LEDs
+    for (var i = 1; i <= 8; i++) {
+        engine.makeConnection("[Channel1]", "hotcue_"+i+"_enabled", MC7000.HotCueLED);
+        engine.makeConnection("[Channel2]", "hotcue_"+i+"_enabled", MC7000.HotCueLED);
+        engine.makeConnection("[Channel3]", "hotcue_"+i+"_enabled", MC7000.HotCueLED);
+        engine.makeConnection("[Channel4]", "hotcue_"+i+"_enabled", MC7000.HotCueLED);
+    }
+
+    // Sampler Mode LED indicator
+    for (i = 1; i <= 8; i++) {
+        engine.makeConnection("[Sampler"+i+"]", "track_loaded", MC7000.SamplerLED);
+        engine.makeConnection("[Sampler"+i+"]", "play", MC7000.SamplerLED);
+    }
+
+    // Sampler Volume Control
+    MC7000.samplerLevel = function(channel, control, value) {
+        // check if the Sampler Volume is at Zero and if so hide the sampler bank
+        if (value > 0x00) {
+            engine.setValue("[Samplers]", "show_samplers", true);
+        } else {
+            engine.setValue("[Samplers]", "show_samplers", false);
+        }
+        // get the Sampler Rows opened with its details
+        engine.setValue("[SamplerRow1]", "expanded", true);
+        engine.setValue("[SamplerRow2]", "expanded", true);
+
+        //control up to 16 sampler volumes with the one knob on the mixer
+        for (var i = 1; i <= 16; i++) {
+            engine.setValue("[Sampler"+i+"]", "pregain", script.absoluteNonLin(value, 0, 1.0, 4.0));
+        }
+    };
+
+    // The SysEx message to send to the controller to force the midi controller
+    // to send the status of every item on the control surface.
+    var ControllerStatusSysex = [0xF0, 0x00, 0x20, 0x7F, 0x03, 0x01, 0xF7];
+
+    // After midi controller receive this Outbound Message request SysEx Message,
+    // midi controller will send the status of every item on the
+    // control surface. (Mixxx will be initialized with current values)
+    midi.sendSysexMsg(ControllerStatusSysex, ControllerStatusSysex.length);
+};
+
+/* CONSTRUCTOR FOR DECK OBJECT */
+MC7000.Deck = function() {
```

Comment:
What is this "Deck object" for? The constructor just assigns global functions, the created (empty) object is never used. You can just remove this.

Reply:
to be honest I have no idea - as mentioned before, never coded, this is my first project as I was looking for a challenge. Well I got one... Reading through the WIKI pages and other mappings, especially the Denon MC4000 and Pioneer SX2. Started with that mapping from scratch and combined a lot of things that I found here and there while no one could efficiently help me. Except @Be-ing who sometimes pointed me. Anyhow, I started with that "MC7000.Deck" but the "this.Something" did not mean anything to me. Yes I did read through Mozilla javascript howtos and explanation, still does not mean anything to me. The first items started to work but while I added thongs then I got several mysterious errors for lines that worked before so I gave up and replaced all "this." with "MC7000." which was just working well by result. Never thought to delete MC7000.Deck = function() but I will try.